framework/BaseYii.php

```diff
@@ -345,23 +345,23 @@ public static function createObject($type, array $params = [])
             return static::$container->get($type, $params);
         }
 
-        if (is_array($type) && isset($type['class'])) {
-            $class = $type['class'];
-            unset($type['class']);
-            return static::$container->get($class, $params, $type);
-        }
-
-        if (is_array($type) && isset($type['__class'])) {
-            $class = $type['__class'];
-            unset($type['__class']);
-            return static::$container->get($class, $params, $type);
-        }
-
         if (is_callable($type, true)) {
             return static::$container->invoke($type, $params);
         }
 
         if (is_array($type)) {
+            if (isset($type['__class'])) {
+                $class = $type['__class'];
+                unset($type['__class'], $type['class']);
```

Comment:
Why need to unset `class` key?

Reply:
Lot of Yii component has another component with default config ['class' => DefaultClass]. This will lead to error `setting unknown property class`.